sympy/functions/combinatorial/numbers.py

```diff
@@ -2292,3 +2292,126 @@ def ok(x):
+#                                                                            #
+#                           Padovan numbers                                  #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class padovan(Function):
+    r"""
+    In number theory, the Padovan sequence is the sequence of integers P(n)
+    defined by the initial values `P_0 = 1`, `P_1 = 1` , `P_2 = 1` and the
+    the two-term recurrence `P_n = P_{n-2} + P_{n-3}`.
+
+    * ``padovan(n)`` gives the `n^{th}` Padovan number, `P_n`
+
+    Examples
+    ========
+
+    >>> from sympy import padovan
+
+    >>> padovan.is_padovan(2)
+    True
+    >>> [padovan(x) for x in range(0,11)]
+    [1, 1, 1, 2, 2, 3, 4, 5, 7, 9, 12]
+    >>> padovan.find_first_n_padovan_numbers(8)
+    [1, 1, 1, 2, 2, 3, 4, 5]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Padovan_sequence
+    .. [2] https://mathworld.wolfram.com/PadovanSequence.html
+
+    """
+
+    @staticmethod
+    def is_padovan(n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            False
+        if n>0:
+            if n == 1 :
+                return True
+
+            sn0 = 1
+            sn1 = 1
+            sn2 = 1
+            while sn2 < n :
+                b = sn0 + sn1
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+
+            if Eq(sn2, n, evaluate=True):
+                return True
+            else:
+                return False
+
+        if n<=0:
+            if n in(0,-1):
+                return True
+
+            sn0 = 1
+            sn1 = 0
+            sn2 = 1
+            while sn2 > n :
+                b = sn0 - sn2
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+
+            if Eq(sn2, n, evaluate=True):
+                return True
+            else:
+                return False
+
+    @staticmethod
+    def find_first_n_padovan_numbers(n):
+        # Returns first n non-negative numbers from the padovan sequence
+        try:
+            n = as_int(n)
+        except ValueError:
+            raise ValueError('The provided number must be a positive integer')
+        if n < 0:
+            raise ValueError('The provided number must be a positive integer')
+        padovan = [1]
+        if n > 1:
+            padovan.append(1)
+
+        sn0 = 1
+        sn1 = 1
+        sn2 = 1
+        i = 2
+        while i < n :
+            padovan.append(sn2)
+            b = sn0 + sn1
+            sn0 = sn1
+            sn1 = sn2
+            sn2 = b
+            i = i + 1
+
+        return padovan
+
+    @classmethod
+    def eval(cls, n):
+        if n is S.Infinity:
+            return S.Infinity
+
+        if n in (0,1,2):
+            return 1
+        if n.is_Integer and n>0:
+            return padovan(n - 2) + padovan(n - 3)
```

Comment:
You might want to use memoization here like `tribonacci` uses instead of directly calling `padovan(n-2)` and `padovan(n-3)`. I'm not too familiar with this module so maybe someone else should confirm that this would be useful.

Reply:
> You might want to use memoization here like `tribonacci` uses instead of directly calling `padovan(n-2)` and `padovan(n-3)`. 

Yeah even I was split ,between using `padovan(n-2) + padovan(n-3) ` or creating a  recurrence memo . I'm not sure what works better but looking at the module it seems the typical approach is using memoization .

Reply:
> it seems the typical approach is using memoization .
Yes, that's what I thought as well. Memoization could probably end up being more efficient here. You could perhaps try timing both methods and see which one is actually faster.

Reply:
@NikhilSDate
 I have added a commit with the recurrence memo .However I had trouble in writing a common recurrence memo for both, when n is positive and when n is negative (because the recurrence relation in the two is different ).Currently I have added code for when n is positive .Can you review it and also improvise one where n (when negative) case is added with it !!
Recurrence relations -
n is positive` f(n) = f(n-2) + f(n-3)`
n is negative `f(n) = f(n+3) - f(n+1)`

Reply:
I think you will need a separate recurrence for the negative case.

Reply:
But then wouldn't I have to define another name !? I believe it is not user friendly if I have to use different func names something like `padovan_pos(x)` for positive and `padovan_neg(x)` for negative.

Reply:
> I think you will need a separate memoization for the negative case.

I will look into this !

Reply:
You can do it similar to how it's handled in `tribonacci` where there is a private function that is memoized which is called by `eval`.  Here you could have two private functions called by `eval` for the two separate cases.

Reply:
@oscarbenjamin  
Could you look into this .I am not sure whether two different recurrence memo's can be created expected to be working with the same defined name `padovan()` .I have attempted to handle it but that code only gives correct results for n>0 (positive n) .Rest all looks good, once this is solved I think this would be good to merge !

Reply:
@praneethratna I would like some hint/assistance on this .I'm not sure if 2 separate recurrence memos can be used under the same name .You could refer to my comment above as reference for what I am trying to improve upon.

Reply:
I'm also not sure of this but i think we cannot use two recurrence memos under the same name.

Reply:
Padovan series for negative numbers is essential ,I would not like to neglect it .For now should I go with the earlier code only ?
 ```
if n in (0,1,2):
            return 1
 if n.is_Integer and n>0:
            return padovan(n - 2) + padovan(n - 3)
```